pkg/reconciler/ingress/ingress_test.go

```diff
@@ -2258,18 +2279,45 @@ func TestReconcileProbingOffClusterGateway(t *testing.T) {
 		}, servicesAndEndpoints...),
 		WantErr: true,
 		WantStatusUpdates: []clientgotesting.UpdateActionImpl{
-			//{Object: ing(withBasicSpec, withGatewayAPIclass, withFinalizer, makeItReadyOffClusterGateway)},
-			{Object: ing(withBasicSpec, withGatewayAPIClass, withFinalizer, func(i *v1alpha1.Ingress) {
-				i.Status.InitializeConditions()
-				i.Status.MarkLoadBalancerNotReady()
-				i.Status.MarkNetworkConfigured()
-				i.Status.MarkIngressNotReady("ReconcileIngressFailed", "Ingress reconciliation failed")
-			}),
-			},
+			{Object: ing(
+				withBasicSpec,
+				withGatewayAPIClass,
+				withFinalizer,
+				func(i *v1alpha1.Ingress) {
+					i.Status.InitializeConditions()
+					i.Status.MarkLoadBalancerNotReady()
+					i.Status.MarkNetworkConfigured()
+					i.Status.MarkIngressNotReady("ReconcileIngressFailed", "Ingress reconciliation failed")
+				},
+			)},
 		},
 		WantEvents: []string{
-			Eventf(corev1.EventTypeWarning, "InternalError", `Gateway "istio-system/istio-gateway" does not have an address in status`),
+			Eventf(corev1.EventTypeWarning, "InternalError", `no address found in status of Gateway istio-system/istio-gateway`),
 		},
+	}, {
+		Name: "gateway doesn't exist",
+		Key:  "ns/name",
+		Ctx: withStatusManager(&fakeStatusManager{
+			FakeDoProbes: func(context.Context, status.Backends) (status.ProbeState, error) {
+				return status.ProbeState{Ready: true}, nil
+			},
+			FakeIsProbeActive: func(types.NamespacedName) (status.ProbeState, bool) {
+				return status.ProbeState{Ready: true}, true
+			},
+		}),
+		Objects: append([]runtime.Object{
+			ing(withBasicSpec, withGatewayAPIclass, withFinalizer, withInitialConditions),
+			httpRoute(t, ing(withBasicSpec, withGatewayAPIclass), httpRouteReady),
+		}, servicesAndEndpoints...),
+		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{Object: ing(
+			withBasicSpec,
+			withGatewayAPIClass,
+			withFinalizer,
+			func(i *v1alpha1.Ingress) {
+				i.Status.InitializeConditions()
+				i.Status.MarkLoadBalancerNotReady()
```

Comment:
this should be loadbalancerfailed

Reply:
hmm weird. when I change it to that, the test fails. It is hitting my code that sets the LB as failed, but for some reason that isn't sticking

Reply:
ah, turns out I was calling MarkLoadBalancerNotReady after that point and it was overwriting it